pkg/cloud/services/ec2/instances.go

```diff
@@ -209,11 +209,19 @@ func (s *Service) CreateInstance(scope *scope.MachineScope, userData []byte) (*i
 	input.SecurityGroupIDs = append(input.SecurityGroupIDs, ids...)
 
 	// Pick SSH key, if any.
-	input.SSHKeyName = aws.String(defaultSSHKeyName)
-	if scope.AWSMachine.Spec.SSHKeyName != "" {
-		input.SSHKeyName = aws.String(scope.AWSMachine.Spec.SSHKeyName)
-	} else if scope.AWSCluster.Spec.SSHKeyName != "" {
-		input.SSHKeyName = aws.String(scope.AWSCluster.Spec.SSHKeyName)
+	// If SSHKeyName WAS NOT provided (nil) then use the defaultSSHKeyName
+	// If SSHKeyName WAS provided *but is an empty string*, do not set a key pair
+	// Otherwise (SSHKeyName WAS provided and is NOT an empty string), use the provided key pair name
+	if scope.AWSMachine.Spec.SSHKeyName != nil {
+		if *scope.AWSMachine.Spec.SSHKeyName != "" {
+			input.SSHKeyName = scope.AWSMachine.Spec.SSHKeyName
+		}
+	} else if scope.AWSCluster.Spec.SSHKeyName != nil {
+		if *scope.AWSCluster.Spec.SSHKeyName != "" {
+			input.SSHKeyName = scope.AWSCluster.Spec.SSHKeyName
+		}
+	} else {
+		input.SSHKeyName = aws.String(defaultSSHKeyName)
 	}
```

Comment:
I tried to make this a bit cleaner looking, but I'm not entirely convinced I did :joy:
```suggestion
	input.SSHKeyName := scope.AWSMachine.Spec.SSHKeyName
	if input.SSHKeyName == nil {
		if scope.AWSCluster.Spec.SSHKeyName != nil {
			input.SSHKeyName := scope.AWSCluster.Spec.SSHKeyName
		} else {
			input.SSHKeyName = defaultSSHKeyName
		}
	} 
```